src/frontend/web_application/src/scenes/ContactBook/presenter.jsx

```diff
@@ -1,95 +1,80 @@
-  {"contact": {"addresses": [], "privacy_features": {}, "phones": [], "contact_id": "98o3907a-eefc-6724-bv28-85d197bb", "date_insert": "2016-05-09T15:01:42.381000", "identities": [], "user_id": "344489c3-olds-4e41-b490-5f4dd317aa53", "title": "Animatronio", "additional_name": null, "date_update": null, "organizations": [], "ims": [], "given_name": "Animatronio", "name_prefix": "", "tags": [], "deleted": 0, "privacy_index": 0, "tags": ['Robots'], "infos": {}, "emails": [{"email_id": "93f03145-4398-4bd4-9bd5-29020001", "is_primary": 0, "date_update": null, "label": null, "address": "animatronio@caliopen.local", "date_insert": "2016-05-09T15:01:42.116000", "type": "other"}], "family_name": null, "name_suffix": null, "avatar": "avatar.png", "public_keys": []}},
-  /* eslint-enable */
-];
+const defaultSortView = 'title';
 
 function getOrderedContacts(contactList, sortView, sortDir) {
-  const sortContacts = contactList.filter(c => c.contact[sortView] != null);
-  const sortNullContacts = contactList.filter(c => c.contact[sortView] == null);
+  const sortContacts = contactList.filter(contact => contact[sortView] !== null);
+  const sortNullContacts = contactList.filter(contact => contact[sortView] == null);
 
   if (sortDir === 'ASC') {
     sortContacts.sort((a, b) =>
-      a.contact[sortView].localeCompare(b.contact[sortView]));
+      a[sortView].localeCompare(b[sortView]));
   }
   if (sortDir === 'DESC') {
     sortContacts.sort((a, b) =>
-      b.contact[sortView].localeCompare(a.contact[sortView]));
+      b[sortView].localeCompare(a[sortView]));
   }
 
   return sortNullContacts.concat(sortContacts);
 }
 
 function getFilteredContacts(contactList, activeTag) {
-  let filteredContacts = contactList.filter(c => c.contact.tags.includes(activeTag));
+  let filteredContacts = contactList.filter(contact => contact.tags.includes(activeTag));
   if (activeTag === '') { filteredContacts = contactList; }
 
   return filteredContacts;
 }
 
 class ContactBook extends Component {
+  static propTypes = {
+    requestContacts: PropTypes.func.isRequired,
+    loadMoreContacts: PropTypes.func.isRequired,
+    contacts: PropTypes.arrayOf(PropTypes.shape({})),
+    isFetching: PropTypes.bool,
+    hasMore: PropTypes.bool,
+    __: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    contacts: [],
+    isFetching: false,
+    hasMore: false,
+  };
 
   constructor(props) {
     super(props);
     this.state = {
-      contactList: [],
       tags: [],
       activeTag: '',
-      sortDir: '',
-      sortView: '',
-      isFetching: true,
+      sortDir: defaultSortDir,
+      sortView: defaultSortView,
     };
+    this.loadMore = this.loadMore.bind(this);
   }
 
   componentDidMount() {
-    this.loadTags();
+    this.props.requestContacts();
     setTimeout(() => {
-      this.loadContacts();
+      this.loadTags();
```

Comment:
ok, here is the problem: `loadTags` does not fetch anything, it uses contacts to calc all tags.

What I think: 
* first rename `loadTags` into `initializeTags` (or calcT.. or extractTa... or ...), it will sounds more accurate

The first time you try to calc tags, there is no contacts since you just fetch its (fortunately, the request arrives within 2ms so you were able to extract the tags, but it won't work if the response take longer).

* Then, you just need to move this call in the [`
componentWillReceiveProps`](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops): every time `props.contacts` change, it will recalc tags